packages/ember-glimmer/tests/integration/helpers/yield-test.js

```diff
@@ -0,0 +1,208 @@
+    this.registerComponent('inner-comp', { template: '{{#outer-comp}}Block{{/outer-comp}}' });
+
+    this.render('{{inner-comp}}');
+    this.assertText('Block');
+
+    this.assertStableRerender();
+  }
+
+  ['@test templates should yield to block, when the yield is embedded in a each helper']() {
+    let list = [1, 2, 3];
+
+    this.registerComponent('outer-comp', { template: '{{#each list as |item|}}{{yield}}{{/each}}' });
+
+    this.render('{{#outer-comp list=list}}Hello{{/outer-comp}}', { list: list });
+    this.assertText('HelloHelloHello');
+
+    this.assertStableRerender();
+
+    this.runTask(() => set(this.context, 'list', [4, 5]));
+    this.assertText('HelloHello');
+
+    this.runTask(() => set(this.context, 'list', list));
+    this.assertText('HelloHelloHello');
+  }
+
+  ['@test templates should yield to block, when the yield is embedded in a if helper']() {
+    this.registerComponent('outer-comp', { template: '{{#if boolean}}{{yield}}{{/if}}' });
+
+    this.render('{{#outer-comp boolean=boolean}}Hello{{/outer-comp}}', { boolean: true });
+    this.assertText('Hello');
+
+    this.assertStableRerender();
+
+    this.runTask(() => set(this.context, 'boolean', false));
+    this.assertText('');
+
+    this.runTask(() => set(this.context, 'boolean', true));
+    this.assertText('Hello');
+  }
+
+  ['@test simple curlies inside of a yielded clock should work when the yield is nested inside of another view']() {
+    this.registerComponent('kiwi-comp', { template: '{{#if falsy}}{{else}}{{yield}}{{/if}}' });
+
+    this.render('{{#kiwi-comp}}{{text}}{{/kiwi-comp}}', { text: 'ohai' });
+    this.assertText('ohai');
+
+    this.assertStableRerender();
+
+    this.runTask(() => set(this.context, 'text', 'portland'));
+    this.assertText('portland');
+
+    this.runTask(() => set(this.context, 'text', 'ohai'));
+    this.assertText('ohai');
+  }
+
+  ['@test nested simple curlies inside of a yielded block should work when the yield is nested inside of another view']() {
+    this.registerComponent('parent-comp', { template: '{{#if falsy}}{{else}}{{yield}}{{/if}}' });
+    this.registerComponent('child-comp', { template: '{{#if falsy}}{{else}}{{text}}{{/if}}' });
+
+    this.render('{{#parent-comp}}{{child-comp text=text}}{{/parent-comp}}', { text: 'ohai' });
+    this.assertText('ohai');
+
+    this.assertStableRerender();
+
+    this.runTask(() => set(this.context, 'text', 'portland'));
+    this.assertText('portland');
+
+    this.runTask(() => set(this.context, 'text', 'ohai'));
+    this.assertText('ohai');
+  }
+
+  // no data to do INUR
+  ['@test yielding to a non-existent block is not an error']() {
+    this.registerComponent('yielding-comp', { template: '<div class="yielding">{{yield}}</div>' });
+    this.registerComponent('outer-comp', { template: '{{yielding-comp}}' });
+
+    this.render('{{outer-comp}}');
+
+    equal(this.$('div.yielding').length, 1);
```

Comment:
@chadhietala the [`nthChild`](https://github.com/emberjs/ember.js/blob/bdb81c40f4ca21b438e3ae39f6056eea13b43f3f/packages/ember-glimmer/tests/utils/abstract-test-case.js#L107-L121) method seems to be more of a `nthSibling` method. 

The DOM that this test produces is 

```html
<div id='ember342'>
   <div id='ember344'>
      <div class='yielding'></div>
   </div>
</div>
```

Hence the your assertion suggestion will not work. 

Is there a need for a help to asset deep into a DOM tree?

I have updated to the test to be simpler. [Here](https://github.com/kiwiupover/ember.js/blob/glimmer-yield-tests/packages/ember-glimmer/tests/integration/helpers/yield-test.js#L92-L101)